source/common/protobuf/utility.cc

```diff
@@ -645,12 +645,18 @@ ProtobufWkt::Value ValueUtil::listValue(const std::vector<ProtobufWkt::Value>& v
 
 namespace {
 
-void validateDuration(const ProtobufWkt::Duration& duration, int64_t max_seconds_value) {
-  if (duration.seconds() < 0 || duration.nanos() < 0) {
-    throw DurationUtil::OutOfRangeException(
-        fmt::format("Expected positive duration: {}", duration.DebugString()));
+absl::Status validateDurationNoThrow(const ProtobufWkt::Duration& duration,
+                                     int64_t max_seconds_value) {
+  if (duration.seconds() < 0 || duration.nanos() < 0 || duration.nanos() > 999999999 ||
+      duration.seconds() > max_seconds_value) {
+    ENVOY_LOG_MISC(warn, "Duration out-of-range: {}", duration.DebugString());
+    return absl::OutOfRangeError("Duration out-of-range");
```

Comment:
nit: return the explicit error message.

Reply:
Could you please elaborate "return the explicit error message"? Do you mean make the error message contain more details?

Reply:
Something in the lines of:
```suggestion
    return absl::OutOfRangeError(fmt::format("Duration out-of-range: {}", duration.DebugString()));
```

Reply:
thanks, done!